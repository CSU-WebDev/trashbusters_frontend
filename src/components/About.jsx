import React from 'react'
import { Container, Row, Col } from 'react-bootstrap'

export default function About() {
    return (
        <>
        <Container fluid id='about-container' className="box">
                <Row>
                    <Col>
                        <h3 className="text-center">About</h3>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <p id='about-paragraph' data-testid='about-paragraph'>Litter is <i>never</i> fun to look at. Join the fight and help bust some trash! Below is our 
                            trash map. You can use it to mark areas on the map where you have spotted litter. Be 
                            sure to add a description if you drop a new pin! You can also delete pins to cleanup 
                            sites once all trash has been removed. Thank you for making our community more beautiful!
                        </p>
                    </Col>
                </Row>
        </Container>
        </>
    )
}
