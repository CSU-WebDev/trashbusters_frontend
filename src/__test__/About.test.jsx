import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import About from '../components/About';

describe('About component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders About heading', () => {
    render(<About />);
    expect(screen.getByText('About')).toBeInTheDocument();
  });

  test('renders about paragraph with correct content', () => {
    render(<About />);
    const aboutParagraph = screen.getByTestId('about-paragraph');
    expect(aboutParagraph).toHaveTextContent(
      /Litter is never fun to look at. Join the fight and help bust some trash!/
    );
  });
});

  