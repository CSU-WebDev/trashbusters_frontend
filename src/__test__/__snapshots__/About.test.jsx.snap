// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box container-fluid"
    data-testid="about-container"
    id="about-container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="text-center"
        >
          About
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <p
          data-testid="about-paragraph"
          id="about-paragraph"
        >
          Litter is 
          <i>
            never
          </i>
           fun to look at. Join the fight and help bust some trash! Below is our trash map. You can use it to mark areas on the map where you have spotted litter. Be sure to add a description if you drop a new pin! You can also delete pins to cleanup sites once all trash has been removed. Thank you for making our community more beautiful!
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
